core/src/main/java/io/grpc/ManagedChannelBuilder.java

```diff
@@ -279,6 +280,18 @@ public T maxInboundMessageSize(int max) {
     return thisT();
   }
 
+  /**
+   * Sets the {@link Proxy} to use when resolving proxy parameters for outgoing
+   * connections.
+   *
+   * @param proxy the proxy to use
+   * @return this
+   */
+  public T proxy(Proxy proxy) {
```

Comment:
`@ExperimentalApi`.

This might be a reasonable case to throw. If someone specifies a SOCKS proxy for instance... Dunno.